infra/auth0/main.tf

```diff
@@ -0,0 +1,604 @@
+
+  colors {
+    # Online-orange
+    primary = "#F9B759"
+    # online-blue
+    page_background = "#0D5474"
+  }
+}
+
+resource "auth0_resource_server" "online" {
+  allow_offline_access                            = true
+  enforce_policies                                = true
+  identifier                                      = "https://online.ntnu.no"
+  name                                            = "Online"
+  signing_alg                                     = "RS256"
+  signing_secret                                  = null
+  skip_consent_for_verifiable_first_party_clients = true
+  token_dialect                                   = "access_token_authz"
+  token_lifetime                                  = 86400
+  token_lifetime_for_web                          = 7200
+  verification_location                           = null
+}
+
+# TODO: feide-log-in
+# resource "auth0_connection" "feide" {
+#   display_name         = null
+#   is_domain_connection = false
+#   metadata             = {}
+#   name                 = "FEIDE"
+#   realms               = ["FEIDE"]
+#   show_as_button       = null
+#   strategy             = "oauth2"
+#   options {
+#     allowed_audiences      = []
+#     api_enable_users       = false
+#     auth_params            = {}
+#     authorization_endpoint = "https://auth.dataporten.no/oauth/authorization"
+#     token_endpoint         = "https://auth.dataporten.no/oauth/token"
+#     client_id              = var.FEIDE_CLIENT_ID
+#     client_secret          = var.FEIDE_CLIENT_SECRET
+#     scopes                 = ["email", "groups", "openid", "phone_number", "profile", "userid-feide"]
+#     scripts = {
+#       fetchUserProfile = file("js/fetchUserProfile.js")
+#     }
+#   }
+# }
+
+# resource "auth0_connection_clients" "feide" {
+#   connection_id = auth0_connection.feide.id
+#   enabled_clients = [
+#     auth0_client.onlineweb_frontend.client_id,
+#     auth0_client.onlineweb4.client_id,
+#     auth0_client.monoweb_web.client_id,
+#     auth0_client.monoweb_dashboard.client_id,
+#     auth0_client.vengeful_vineyard_frontend.client_id,
+#   ]
+# }
+
+# resource "auth0_action" "klassetrinn_autoconf" {
+#   name = "Klassetrinn Autoconf"
+#   code = file("js/klassetrinn.js")
+
+#   supported_triggers {
+#     id = "post_signup"
+#     version = "v3"
+#   }
+# }
+
+resource "auth0_client" "vengeful_vineyard_frontend" {
+  app_type = "regular_web"
+  callbacks = flatten([
+    terraform.workspace == "dev" ? ["http://localhost:3000"] : [],
+    terraform.workspace == "prd" ? [] : [],
+  ])
+  grant_types                   = ["authorization_code", "refresh_token"]
+  name                          = "Vengeful Vineyard${terraform.workspace == "dev" ? " Dev" : ""}"
+  organization_require_behavior = "no_prompt"
+  is_first_party                = true
+  oidc_conformant               = true
+  
+  jwt_configuration {
+    alg = "RS256"
+  }
+}
```

Comment:
We can leave them here for now, but I think that in the future, the other apps themselves should register their own clients

At that point we should probably abstract over with Terraform modules too.